packages/jellyfish-api-core/src/category/masternode.ts

```diff
@@ -141,15 +141,49 @@ export class Masternode {
   }
 
   /**
-   * Get information about governance variable
+   * Set special governance variables with activation height specified
    *
+   * @param {Record<string, any>} input json object
+   * @param {number} activationHeight
+   * @return {Promise<string>} hash
+   *
+   */
+  async setGovWithHeight (input: Record<string, any>, activationHeight: number): Promise<string> {
+    const currentHeight: number = await this.client.call('getblockcount', [], 'number')
+    if (activationHeight < currentHeight) {
+      throw new Error('GovVar activation height must be higher than current block height')
+    }
+    return await this.client.call('setgovheight', [input, activationHeight], 'number')
+  }
+
+  /**
+   * Pre Fort Canning: Get information about governance variable
+   *
+   * @deprecated use {@link getGovWithHeight} for post Fort Canning hardfork
    * @param {string} name governance name
    * @return {Promise<Record<string, any>} governance information as json object
    */
   async getGov (name: string): Promise<Record<string, any>> {
     return await this.client.call('getgov', [name], 'bignumber')
   }
 
+  /**
+   * Post Fort Canning: Get information about governance variable
+   * To replace {@link getGov}
+   *
+   * @param {string} name governance name
+   * @return {Promise<Array<Record<string, any>>} governance information as json object
+   * @example
+   * {
+   *   "LP_SPLITS": { "3": BigNumber, "4": BigNumber },
+   *   "5000": { "3": BigNumber, "4": BigNumber }, // value after block height 5000
+   *   "8000": { "3": BigNumber, "4": BigNumber } // value after block height 8000
+   * }
+   */
+  async getGovWithHeight (name: string): Promise<Array<Record<string, any>>> {
```

Comment:
Due to regression in https://github.com/DeFiCh/jellyfish/pull/753 we can't deprecate getGov, we have to replace it and fix all the tests that broke due to it. This change has to be moved to #753 due to various regression

Reply:
Porting changes by hand, no more multiple versions.

Reply:
was checking why we need `Array<Record<string, any>>` in the return object. This is the structure of the return data
```
 [
      { LP_SPLITS: { '3': [BigNumber], '4': [BigNumber] } },
      { '109': { '3': [BigNumber], '4': [BigNumber] } },
      { '114': { '3': [BigNumber], '4': [BigNumber] } }
    ]
```

seems bit odd to me. Why can't  it return `Record<string, any>` type. i.e
```
{
      LP_SPLITS: { '3': [BigNumber], '4': [BigNumber] },
      '109': { '3': [BigNumber], '4': [BigNumber] } ,
      '114': { '3': [BigNumber], '4': [BigNumber] }
}
```
does it need the extra Array?

Reply:
btw we also have a `listgovs` RPC